Add content/en/continuous_deployment.md

Vrukshali-26 opened the pull request:
Signed-off-by: Vrukshali <tvrukshali@gmail.com>

Renamed "continuous_delivery.md" file to "continuous_delivery&continuous_deployment.md"
Added more information about Continuous Deployment and also about Continuous Deployment vs Continuous Delivery.

CathPag commented:
Thanks, @Vrukshali-26. Not sure I understand the change. We already have a definition for both: [Continuous Delivery (CD)](https://glossary.cncf.io/continuous_delivery/) and [Continuous integration (CI)](https://glossary.cncf.io/continuous_integration/). Why add a new CI definition below the current CD definition?

Vrukshali-26 commented:
Ohh Okay! I will change.

Vrukshali-26 commented:
I have made the changes @CathPag Can you have a look?

CathPag commented:
Sorry, I'm really confused. There are two CD definitions here. Could you please explain the problem you are trying to fix? 

Also, you changed continuous **delivery** to **deployment**. There is an entire discussion on whether CD refers to delivery or deployment and what each means. We could add a small paragraph to that. If you feel we should change the definition from delivery to deployment, we may need to start a discussion.

Vrukshali-26 commented:
@CathPag let me clear this confusion. 
So the issue I was assigned was to explain continuous deployment. We already had a continuous delivery explanation.
1. At first, I added the information continuous deployment to continuous delivery file.
2. But now I created a different file continuous deployment one and added the content.
3. So now we have two different files. {Continuous delivery and continuous deployment}
So how it goes is:
Continuous Integration (CI) --> Continuous Delivery (CD) --> Continuous Deployment (CD)

Hope I'm able to explain what I did🙂

Vrukshali-26 commented:
I have made the changes @CathPag!

jihoon-seo commented on content/en/continuous deployment.md:
```diff
@@ -0,0 +1,20 @@
+---
```
Please consider to rename this file to `continuous_deployment.md`, for the consistency. 😊

([Files in `content/en/`](https://github.com/cncf/glossary/tree/main/content/en))

Vrukshali-26 replied:
I have made the changes @jihoon-seo!

jihoon-seo reviewed (commented):
@Vrukshali-26 Thank you for your contribution! 😊

And PTAL! 😊

CathPag commented:
Sorry, @Vrukshali-26, could you remove continuous delivery from this PR? It should be one term per entry / PR. Continuous delivery should be its own separate entry, so please update the PR name also. Once the PR is cleanly separated, we focus on the content.

JasonMorgan commented:
@Vrukshali-26 Thanks so much for this, I believe it's a fantastic addition to the glossary. That being said there are some changes that will be required to go forward. Please feel free to reach out to me directly on slack if you would like some help. 

Thanks!

Vrukshali-26 commented:
@CathPag I have updated the PR name, anything else to change?

Vrukshali-26 commented:
@jihoon-seo I can't figure out how can I undo those changes in continuous_delivery.md file, can you help me with that?

jihoon-seo commented:
On your `add-content-continuous-deployment` branch of your dev environment,

1. Check your 4 commits using `git log` or other tools:
![image](https://user-images.githubusercontent.com/46767780/155242699-5424bfa7-be2e-4cfa-acf8-46648a6a954e.png)
![image](https://user-images.githubusercontent.com/46767780/155242753-054cdc90-e17f-45ad-a950-37b52590d6aa.png)

2. Reset your 4 commits, keeping your changes "staged".
`git reset --soft HEAD~4`
![image](https://user-images.githubusercontent.com/46767780/155242958-00c5ef7d-46a1-49c6-b4b0-c684f3cf44ff.png)
![image](https://user-images.githubusercontent.com/46767780/155243008-04408d81-afc8-49ea-aa11-b2990ac6388a.png)

3. Unstage and discard the change on `content/en/continuous_delivery.md` file.
`git reset HEAD content/en/continuous_delivery.md`
`git checkout -- content/en/continuous_delivery.md`

4. Make sure that the change on `content/en/continuous_deployment.md` is the only one staged.
`git status`
![image](https://user-images.githubusercontent.com/46767780/155243491-bb2471bf-2d77-4f31-8f6d-f88b4383df60.png)

5. Create a new commit with an appropriate commit message:
`git commit -m "Add 'Continuous Deployment'"`
`git log`
![image](https://user-images.githubusercontent.com/46767780/155243769-580debff-f8ad-40a0-a5b8-df1ea03c24f4.png)

6. Force-push the branch to your forked GitHub repo.
`git push origin add-content-continuous-deployment --force`
Then this PR will be updated accordingly.

Hope this helps! 😊

Vrukshali-26 commented:
I did it @jihoon-seo
Thank you for help🤠

CathPag commented:
> @CathPag I have updated the PR name, anything else to change?

Yes, the text needs some adjustments. Jason asked if you could DM him so he can explain in more detail. Mind reaching out to him on Slack?

Vrukshali-26 commented:
> > @CathPag I have updated the PR name, anything else to change?
> 
> Yes, the text needs some adjustments. Jason asked if you could DM him so he can explain in more detail. Mind reaching out to him on Slack?

I have texted him on slack!

Vrukshali-26 commented:
> @Vrukshali-26 Thanks so much for this, I believe it's a fantastic addition to the glossary. That being said there are some changes that will be required to go forward. Please feel free to reach out to me directly on slack if you would like some help.
> 
> Thanks!

I have DMed you on slack!

CathPag commented:
@Vrukshali-26, I talked to Jason and he said that he mentioned the most important points on GitHub. Unfortunately, half of the content is from [this article ](https://www.atlassian.com/continuous-delivery/continuous-deployment/how-to-get-to-continuous-deployment) (word for word) and another half paragraph is from [this one](https://www.indellient.com/blog/whats-the-difference-between-continuous-integration-continuous-delivery-and-continuous-deployment/). 

We do want you to look at how others explain terms but I'm afraid we can't copy it word by word. Try to use your own words, please. Also, make sure the style is consistent. We don't use quotes or numbering.

Wanna give it another try?

CathPag commented:
@Vrukshali-26, just following up on this

Vrukshali-26 commented:
@CathPag @JasonMorgan  I have made the changes!

CathPag commented on content/en/continuous_deployment.md:
```diff
@@ -0,0 +1,14 @@
+Anytime the new changes are released, it causes the entire infrastructure to slow down as they need to manually keep on checking if everything is proper on the production side. If anything goes wrong it may cause huge losses to that industry. So this causes industries to slow down the process of new release.
+
+## How it helps
+Continuous Deployment solves the above problem by automating the process. Rather than making a big change, industry works on making smaller changes and releasing it, so that instead of slowing down the entire process everything goes smoothly because of the small changes and automated process of testing and pushing it to production. Also, the feedback of the customers is faster!!
```
```suggestion
Continuous deployment automates the process of pushing code to production. CI/CD, in general, focuses on releasing smaller changes more often, implementing deployment strategies such as [canary](/canary_deployment/) or [blue-green](/blue_green_deployment/) releases, and ultimately delivering value to customer faster. 
```

CathPag replied:
```suggestion
Continuous deployment automates the process of pushing code to production. CI/CD, in general, focuses on releasing smaller changes more often and implementing deployment strategies such as [canary](/canary_deployment/) or [blue-green](/blue_green_deployment/) releases.
```

CathPag reviewed (commented):
Thank you, @Vrukshali-26! Made some suggestions and a comment regarding the middle section.

JasonMorgan commented:
@Vrukshali-26 I made some significant edits to your entry. Please let me know if you're happy with the changes.

CathPag reviewed (approved):
LGTM
